cmd/veneur/main.go

```diff
@@ -28,14 +29,20 @@ func main() {
 	}
 
 	conf, err := veneur.ReadConfig(*configFile)
+	var exitcode int
 	if err != nil {
+		exitcode = 1
 		if _, ok := err.(*veneur.UnknownConfigKeys); ok {
 			logrus.WithError(err).Warn("Config contains invalid or deprecated keys")
```

Comment:
It seems like there's a design you're going for here, but i'm not seeing it? The benefit of having this flag is that we can run a check earlier in CI, so that we can't get to a crash-loop state because by the time veneur gets deployed, we can be assured that the config works.

Reply:
I think validating that the config will be warning- and error-free makes backwards-compatible upgrades a bit harder: currently, it's possible to install a config file that has keys which the currently-deployed version of veneur doesn't expect; failing the build for those would mean we always have to roll out everything in lockstep.